### ISSUE TITLE
Stuck on "Confirm Transaction" after canceling purchase due to insufficient funds

### ISSUE THREAD

**Issue by franckc** [OriginProtocol/origin#3479]
Repro steps:
 - Mobile app v0.23.18 on iOS.
 - Click purchase on https://shoporigin.com/#/listing/1-000-3136
 - DApp shows "Confirm transaction" modal
 - Buyer has enough ETH to cover the purchase but not the gas fees. So the wallet shows an error about not having enough ETH. Click cancel.
 => DApp remains stuck on the "Confirm transaction" modal.
![2019-09-24 13 39 37](https://user-images.githubusercontent.com/945910/65549110-b81dc280-ded1-11e9-900b-319db9c45e83.png)

**Comment by franckc** [comment]
@micahalcorn I don't think this is tracked by any other issue ?

**Comment by micahalcorn** [comment]
No, I didn't know about this one @franckc. 🙁 We probably shouldn't let you get into this state in the first place. So the "insufficient funds" calculation needs to be gas-aware.

**Comment by micahalcorn** [comment]
Although, I guess we don't know what the gas amount is going to be across various environments. So maybe we need to just better handle the dismissal of this modal. 😕 I assume you saw an "insufficient funds" message in our native card that came up from the bottom @franckc?

**Comment by franckc** [comment]
Correct. The "insufficient card" was native, came from the bottom and got on top of the "Confirm transaction" webview modal.
